### PR TITLE
Homework4: Aleksandr Levdansky

### PR DESCRIPTION
**Pull request by Coromilly**

### CODE REVIEW

File: func_decor.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
+from time import time
+
+registry = {}
+
+
+def wrapper(func):
```
Comment (belovmd) [inline func_decor.py]: usually we call wrapper internal function

File: func_decor.py [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+def get_ranges(lst):
+    """Function "get_ranges".
+
+    This function takes list of unique sorted ascending integer numbers
+    and returns this list rolled up.
+    For example:
+    get_ranges([0, 1, 2, 3, 4, 7, 8, 10]) // "0-4,7-8,10"
+    """
+    pos = 0
+    rolled_list = str(lst[0])
```
Comment (belovmd) [inline func_decor.py]: am I correct you name string variable list?

File: func_decor.py [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    return timer
+
+
+@wrapper
+def spam(*args):
+    return {arg: arg ** 2 for arg in args}  # spam = wrapper(spam)
+
+
+@wrapper
+def get_ranges(lst):
```
Comment (belovmd) [inline func_decor.py:25]: your function doesn't work correctly and has wrong output

**Review by belovmd — COMMENTED** [review]
second task doesn't work correctly. 
Also please address other minor issues

File: func_decor.py (hunk 2 above)
Comment (Coromilly) [reply func_decor.py]: Yep. Y answered in telegram some days ago that output should be a string. The task was to roll out the list.
I'll correct to "roll".

File: func_decor.py (hunk 1 above)
Comment (Coromilly) [reply func_decor.py]: solved.

File: func_decor.py (hunk 3 above)
Comment (Coromilly) [reply func_decor.py:25]: solved.

File: func_decor.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+registry = {}
+
+
+def timer(func):
+    def wrapper(*args, **kwargs):
+        start = time()
+        res = func(*args, **kwargs)
+        stop = time()
+        registry[func.__name__] = (
```
Comment (belovmd) [inline func_decor.py:11]: it would rewrite results after every function call

File: get_ranges.py [hunk 5]
```diff
@@ -0,0 +1,26 @@
+def get_ranges(lst):
```
Comment (belovmd) [inline get_ranges.py:1]: why do we need two same functions? can't we use import?

**Review by belovmd — COMMENTED** [review]
6.5

**Pull request closed without merging**